### PR TITLE
[18] Switch code samples to createRoot

### PR DESCRIPTION
**Pull request by rickhanlonii**
## Overview

Switches in-line code examples to `createRoot`. Will do the `/examples` separately.

### CODE REVIEW

File: content/docs/thinking-in-react.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ To build a static version of your app that renders your data model, you'll want
...
 
-At the end of this step, you'll have a library of reusable components that render your data model. The components will only have `render()` methods since this is a static version of your app. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. If you make a change to your underlying data model and call `ReactDOM.render()` again, the UI will be updated. You can see how your UI is updated and where to make changes. React's **one-way data flow** (also called *one-way binding*) keeps everything modular and fast.
+At the end of this step, you'll have a library of reusable components that render your data model. The components will only have `render()` methods since this is a static version of your app. The component at the top of the hierarchy (`FilterableProductTable`) will take your data model as a prop. If you make a change to your underlying data model and call `render()` again, the UI will be updated. You can see how your UI is updated and where to make changes. React's **one-way data flow** (also called *one-way binding*) keeps everything modular and fast.
```
Comment (sebmarkbage) [inline content/docs/thinking-in-react.md:73]: In this case the `render()` refers to `root.render()` so it's worth having a name for the root variable in the case.

File: content/docs/rendering-elements.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ We call this a "root" DOM node because everything inside it will be managed by R
 
 Applications built with just React usually have a single root DOM node. If you are integrating React into an existing app, you may have as many isolated root DOM nodes as you like.
 
-To render a React element into a root DOM node, pass both to [`ReactDOM.render()`](/docs/react-dom.html#render):
+To render a React element, first create a React root with [`ReactDOM.createRoot()`](/docs/react-dom-client.html#createroot) and pass the element to `root.render()`:
```
Comment (sebmarkbage) [inline content/docs/rendering-elements.md:37]: pass the *React* element.

The DOM element to createRoot, the React element to render.

File: content/docs/typechecking-with-proptypes.md [hunk 3]
```diff
@@ -173,10 +173,8 @@ Greeting.defaultProps = {
 };
 
 // Renders "Hello, Stranger":
-ReactDOM.render(
-  <Greeting />,
-  document.getElementById('example')
-);
+const root = ReactDOM.render(document.getElementById('example'));
```
Comment (sebmarkbage) [inline content/docs/typechecking-with-proptypes.md:176]: This should be createRoot

**Comment by rickhanlonii** [comment]
Thanks @sebmarkbage, fixed!

**Review by sebmarkbage — APPROVED** [review]

File: content/docs/components-and-props.md [hunk 4]
```diff
@@ -111,11 +109,6 @@ function App() {
     </div>
   );
 }
-
```
Comment (harish-sethuraman) [inline content/docs/components-and-props.md:114]: We decided to leave this render intentionally? 

preview: https://reactjs-org-7ht1fcs9w-fbopensource.vercel.app/docs/components-and-props.html#composing-components

docs: https://reactjs.org/docs/components-and-props.html#composing-components

File: content/docs/hello-world.md [hunk 5]
```diff
@@ -8,11 +8,8 @@ next: introducing-jsx.html
 
 The smallest React example looks like this:
 
-```js
-ReactDOM.render(
-  <h1>Hello, world!</h1>,
-  document.getElementById('root')
-);
+```jsx
+<h1>Hello, world!</h1>
```
Comment (harish-sethuraman) [inline content/docs/hello-world.md:12]: Nice to have:
```suggestion
ReactDOM.createRoot(document.getElementById('root')).render(<h1>Hello, world!</h1>);
```

Preview link: https://reactjs-org-7ht1fcs9w-fbopensource.vercel.app/docs/hello-world.html
Docs: https://reactjs.org/docs/hello-world.html

File: content/docs/reference-glossary.md [hunk 6]
```diff
@@ -41,11 +41,10 @@ React DOM uses camelCase property naming convention instead of HTML attribute na
 
 ```js
 const name = 'Clementine';
-ReactDOM.render(
+root.render(
```
Comment (harish-sethuraman) [inline content/docs/reference-glossary.md:44]:
```js
const root = ReactDOM.createRoot(document.getElementById('root'));
```

File: content/docs/components-and-props.md (hunk 4 above)
Comment (rickhanlonii) [reply content/docs/components-and-props.md:114]: Yeah, this code isn't important/needed for the example

File: content/docs/reference-glossary.md (hunk 6 above)
Comment (rickhanlonii) [reply content/docs/reference-glossary.md:44]: I'll just remove the render part, it's not relevant here, where the focus is on `class` vs `className`

File: content/docs/hello-world.md (hunk 5 above)
Comment (rickhanlonii) [reply content/docs/hello-world.md:12]: Thanks!

**Review by harish-sethuraman — APPROVED** [review]
🚀

**Pull request merged**